pkg/core/message.go

```diff
@@ -110,9 +116,10 @@ type Message struct {
 // Fields such as the state/confirmed do NOT transfer, as these are calculated individually by each member.
 func (m *Message) BatchMessage() *Message {
 	return &Message{
-		Header: m.Header,
-		Hash:   m.Hash,
-		Data:   m.Data,
+		Header:       m.Header,
+		Hash:         m.Hash,
+		Data:         m.Data,
+		Transactions: m.Transactions,
```

Comment:
I'm not sure this meets the constraints above for what should be transferred/immutable in the batch. It's not part of the hash, so there's really no verification on it. It's a convenient way to ensure the far side receives `transactions.related`... if they wanted to recompute it, it would have to be done by the event aggregator, by looking up the transfer/approval and copying the TX ID off of it and updating the message with it.

Reply:
Although... I guess I _could_ put the "related" transaction ID (the one for the token transfer or approval) into the message header. It would require some code shuffling, but it is computed in the same code path as the message sealing. The batch transaction ID would have to be outside the header, as it's computed much later, after sealing.

I guess maybe that would look something like:
```
{
  "header": {
    "txtype": "batch_pin",
    "txparent": {
      "type": "token_transfer",
      "id": "<id for transfer transaction>"
    }
  },
  "txid": "<id for batch pin transaction"
}
```

It's slightly weird in how it's organized, but I don't hate it. It would also solve #1157 in a different way, by adding a new field in the message header that tells you to correlate the message with a transfer or approval.

Reply:
I went ahead and explored this alternate route as #1163... so the saga continues.